# Imports and Requirements
For this project, we need the data sets and some python libraries

In [1]:
import pandas as pd
import numpy as np

## Reading and cleaning the data

In [3]:
df_original = pd.read_csv('./data/chemicals-in-cosmetics.csv')
df = df_original.drop_duplicates()
print('The original data had ', df_original.shape[0], "rows")
print('After removing duplicates, the data has', df.shape[0], "rows")

The original data had  112870 rows
After removing duplicates, the data has  112616 rows


### Sampling the data

In [4]:
df.head()

CDPHId                                       ProductName  CSFId  CSF  \
0       2  ULTRA COLOR RICH EXTRA PLUMP LIPSTICK-ALL SHADES    NaN  NaN   
1       3                        Glover's Medicated Shampoo    NaN  NaN   
2       3                        Glover's Medicated Shampoo    NaN  NaN   
3       4          PRECISION GLIMMER EYE LINER-ALL SHADES �    NaN  NaN   
4       5       AVON BRILLIANT SHINE LIP GLOSS-ALL SHADES �    NaN  NaN   

   CompanyId          CompanyName BrandName  PrimaryCategoryId  \
0          4         New Avon LLC      AVON                 44   
1        338  J. Strickland & Co.  Glover's                 18   
2        338  J. Strickland & Co.  Glover's                 18   
3          4         New Avon LLC      AVON                 44   
4          4         New Avon LLC      AVON                 44   

                     PrimaryCategory  SubCategoryId  ...   CasNumber  \
0    Makeup Products (non-permanent)             53  ...  13463-67-7   
1  Hair Care Products (non-coloring)             25  ...  65996-92-1   
2  Hair Care Products (non-coloring)             25  ...    140-67-0   
3    Makeup Products (non-permanent)             46  ...  13463-67-7   
4    Makeup Products (non-permanent)             52  ...  13463-67-7   

   ChemicalId            ChemicalName  InitialDateReported  \
0           6        Titanium dioxide           06/17/2009   
1           4  Distillates (coal tar)           07/01/2009   
2           5               Estragole           07/01/2009   
3           7        Titanium dioxide           07/09/2009   
4           8        Titanium dioxide           07/09/2009   

  MostRecentDateReported DiscontinuedDate ChemicalCreatedAt ChemicalUpdatedAt  \
0             08/28/2013       02/01/2011        07/09/2009        07/09/2009   
1             07/01/2009              NaN        07/01/2009        07/01/2009   
2             07/01/2009              NaN        07/02/2009        07/02/2009   
3             08/28/2013              NaN        07/09/2009        07/09/2009   
4             08/28/2013       02/01/2011        07/09/2009        07/09/2009   

  ChemicalDateRemoved ChemicalCount  
0                 NaN             1  
1                 NaN             2  
2                 NaN             2  
3                 NaN             1  
4                 NaN             1  

[5 rows x 22 columns]

# Analysis
Let us look at the total number of chemicals that we have in our dataset

In [5]:
df['ChemicalName'].value_counts().size

123

It seems that we have a total of 123 chemicals in our given data

### Trends and Averages
Let us see what is the average number of reported chemicals, as well as the maximum and minimum for each product

In [6]:
df['ChemicalCount'].describe()

count    112616.000000
mean          1.282402
std           0.629696
min           0.000000
25%           1.000000
50%           1.000000
75%           1.000000
max           9.000000
Name: ChemicalCount, dtype: float64

This tells us that there are some products with no reported chemicals, and there arae some with as many as 9 reported chemicals.

However, the average seems to be around 1 chemical

### Removing some bias from our observations

It doesn't make sense that some products have no reported chemicals at all, so let us closely examine what we have

In [7]:
df.loc[df.ChemicalCount==0].head()

CDPHId                ProductName  CSFId  CSF  CompanyId  \
31       24  White Premium Lotion Soap    NaN  NaN        181   
497     333            Gentle Cleanser    NaN  NaN         71   
498     334             Cleansing Foam    NaN  NaN         71   
499     334             Cleansing Foam    NaN  NaN         71   
500     334             Cleansing Foam    NaN  NaN         71   

                      CompanyName BrandName  PrimaryCategoryId  \
31          GOJO Industries, Inc.     GOJO�                 74   
497  Sunrider Manufacturing, L.P.   Kandesn                  6   
498  Sunrider Manufacturing, L.P.   Kandesn                 90   
499  Sunrider Manufacturing, L.P.   Kandesn                 90   
500  Sunrider Manufacturing, L.P.   Kandesn                 90   

            PrimaryCategory  SubCategoryId  ...   CasNumber  ChemicalId  \
31   Personal Care Products            170  ...  68603-42-9          26   
497           Bath Products            159  ...  68603-42-9         350   
498     Skin Care Products              93  ...  68603-42-9         351   
499     Skin Care Products              93  ...  68603-42-9         412   
500     Skin Care Products              93  ...         NaN         453   

                                          ChemicalName  InitialDateReported  \
31                             Cocamide diethanolamine           07/23/2009   
497                            Cocamide diethanolamine           09/02/2009   
498                            Cocamide diethanolamine           09/02/2009   
499                            Cocamide diethanolamine           09/02/2009   
500  Retinol/retinyl esters, when in daily dosages ...           09/02/2009   

    MostRecentDateReported DiscontinuedDate ChemicalCreatedAt  \
31              02/07/2012              NaN        07/23/2009   
497             09/16/2013              NaN        09/02/2009   
498             09/16/2013              NaN        09/02/2009   
499             09/16/2013              NaN        09/03/2009   
500             09/16/2013              NaN        09/03/2009   

    ChemicalUpdatedAt ChemicalDateRemoved ChemicalCount  
31         02/07/2012          12/30/2010             0  
497        09/16/2013          11/23/2009             0  
498        09/03/2009          09/02/2009             0  
499        12/05/2013          09/02/2009             0  
500        12/05/2013          12/05/2103             0  

[5 rows x 22 columns]